ushahidi/platform-client-mzima
# [Outreachy Task Submission] Buttons Must have discernible text for screen readers

# Description
Some of the buttons like the password hide button on the login pages and the share button on the feed pages have no inner text so adding aria-label will help in accessibility ie the filter and the share-modal component.

For the mzima button which is a component used across the site, the earlier implementation does not allow for aria-labels and adding one throws an error. So I added  a span with text to help in screen reader accessibility, then I added a css class that hides the span from the naked eye ie in the registration, collection and search form components.


## Fixes 
This PR fixes the issue: https://github.com/ushahidi/platform/issues/4761 and https://github.com/ushahidi/platform/issues/4763

## Testing
To test: 
- Install [axeDevTools ](https://chromewebstore.google.com/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd)on your chrome browser and scan the pages where the file have been changed - no button must have discernible text error
- remove the aria-label or the span element and scan the page. You will get a 'Button must have discernible text error' You can also highlight to see which icon/button is problematic.
